argh, of course the assertions aren't correct on Windows: https://ci.ros2.org/job/ci_windows/26643/testReport/junit/launch_yaml.test.launch_yaml/test_path_join_substitution/test_path_join_substitution/

```
AssertionError: assert 'file=robot2\...\\v_1.0.xacro' == 'file=robot2/...f/v_1.0.xacro'
  - file=robot2/a/ur df/v_1.0.xacro
  ?            ^ ^     ^
  + file=robot2\a\ur df\v_1.0.xacro
  ?            ^ ^     ^
```

--- a/launch_xml/test/launch_xml/test_path_join_substitution.py
+++ b/launch_xml/test/launch_xml/test_path_join_substitution.py
@@ -15,6 +15,7 @@
 """Test parsing a `PathJoinSubstitution` substitution."""
 
 import io
+from pathlib import Path
 import textwrap
 
 import lark
@@ -52,7 +53,8 @@ def test_path_join_substitution():
     log_info = launch_description.entities[3]
     assert isinstance(log_info, Log)
     assert isinstance(log_info.msg[1], PathJoinSubstitution)
-    assert perform_substitutions(ls.context, log_info.msg) == 'file=robot2/a/ur df/v_1.0.xacro'
+    assert perform_substitutions(ls.context, log_info.msg) == \
+        'file=' + str(Path('robot2', 'a', 'ur df', 'v_1.0.xacro'))
 
 
 def test_path_join_substitution_empty():

--- a/launch_yaml/test/launch_yaml/test_path_join_substitution.py
+++ b/launch_yaml/test/launch_yaml/test_path_join_substitution.py
@@ -15,6 +15,7 @@
 """Test parsing a `PathJoinSubstitution` substitution."""
 
 import io
+from pathlib import Path
 import textwrap
 
 import lark
@@ -56,7 +57,8 @@ def test_path_join_substitution():
     log_info = launch_description.entities[3]
     assert isinstance(log_info, Log)
     assert isinstance(log_info.msg[1], PathJoinSubstitution)
-    assert perform_substitutions(ls.context, log_info.msg) == 'file=robot2/a/ur df/v_1.0.xacro'
+    assert perform_substitutions(ls.context, log_info.msg) == \
+        'file=' + str(Path('robot2', 'a', 'ur df', 'v_1.0.xacro'))
 
 
 def test_path_join_substitution_empty():

